src/api/app/models/package.rb

```diff
@@ -1134,10 +1155,14 @@ def save_file(opt = {})
     content = opt[:file] if opt[:file]
     logger.debug "storing file: filename: #{opt[:filename]}, comment: #{opt[:comment]}"
 
+    Package.verify_file!(self, opt[:filename], opt[:file])
+    unless User.current.can_modify_package?(self)
```

Comment:
I was only refactoring the driver_update_controller to get rid of FrontEndCompat... not the Package model itself... it have to be refactored in other Trello card.